### PR TITLE
Upgrade to docker-client v3.1.0

### PR DESCRIPTION
**Pull request by rohansingh**
This is required for exec health checks to work with Docker 1.7+.

### CODE REVIEW

File: helios-system-tests/src/main/java/com/spotify/helios/system/ResourcesTest.java [hunk 1]
```diff
@@ -99,13 +99,13 @@ public void testClient() throws Exception {
...
-      final ContainerConfig containerConfig =
-          docker.inspectContainer(taskStatus.getContainerId()).config();
+      final HostConfig hostConfig =
+          docker.inspectContainer(taskStatus.getContainerId()).hostConfig();
 
-      assertEquals(MEMORY, containerConfig.memory());
-      assertEquals(MEMORY_SWAP, containerConfig.memorySwap());
-      assertEquals(CPU_SHARES, containerConfig.cpuShares());
-      assertEquals(CPUSET, containerConfig.cpuset());
+//      assertEquals(MEMORY, hostConfig.memory());
```
Comment (davidxia) [inline helios-system-tests/src/main/java/com/spotify/helios/system/ResourcesTest.java]: For some reason, the memory limits aren't being on `docker.createContainer()` [here][1].

  [1]: https://github.com/spotify/helios/pull/590/files#diff-12ef098bf3b4de9a99db3d575871cb50R227

**Comment by rohansingh** [comment]
@igorbernstein

**Comment by gimaker** [comment]
:+1:

**Comment by davidxia** [comment]
@rohansingh yay, finally fixed the tests. See comments inline.

File: helios-system-tests/src/main/java/com/spotify/helios/system/ResourcesTest.java [hunk 2]
```diff
@@ -64,21 +69,18 @@ public void setup() throws Exception {
...
-      assumeTrue(info.memoryLimit());
-      assumeTrue(info.swapLimit());
-    }
   }
 
   @Test
   public void testClient() throws Exception {
+    // Doesn't work on CircleCI because their lxc-driver can't set cpus
+    // See output of `docker run --cpuset-cpus 0-1 busybox true`
+    assumeThat(getenv("CIRCLECI"), isEmptyOrNullString());
```
Comment (davidxia) [inline helios-system-tests/src/main/java/com/spotify/helios/system/ResourcesTest.java]: Doesn't work on circleci.

File: helios-system-tests/src/main/java/com/spotify/helios/system/ResourcesTest.java [hunk 3]
```diff
@@ -99,13 +101,28 @@ public void testClient() throws Exception {
...
+      assertEquals(CPUSET_CPUS, hostConfig.cpusetCpus());
+
+      final Info info = docker.info();
+      final Iterable<String> split = Splitter.on(".").split(docker.version().apiVersion());
+      //noinspection ConstantConditions
+      final int major = Integer.parseInt(Iterables.get(split, 0, "0"));
+      //noinspection ConstantConditions
+      final int minor = Integer.parseInt(Iterables.get(split, 1, "0"));
+
+      // TODO (dxia) This doesn't work on docker < 1.7 ie docker API < 1.19 for some reason.
```
Comment (davidxia) [inline helios-system-tests/src/main/java/com/spotify/helios/system/ResourcesTest.java]: NB

File: helios-services/src/main/java/com/spotify/helios/agent/SyslogRedirectingContainerDecorator.java [hunk 4]
```diff
@@ -75,7 +75,8 @@ public void decorateContainerConfig(Job job, ImageInfo imageInfo,
 
     // If there's no explicit container cmd specified, copy over the one from the image.
     // Only setting the entrypoint causes dockerd to not use the image cmd.
-    if (containerConfig.cmd().isEmpty() && imageConfig.cmd() != null) {
+    if ((containerConfig.cmd() == null || containerConfig.cmd().isEmpty())
```
Comment (davidxia) [inline helios-services/src/main/java/com/spotify/helios/agent/SyslogRedirectingContainerDecorator.java]: For some reason, `cmd()` can return null now.

**Comment by davidxia** [comment]
@rohansingh tests passed for me locally

**Pull request merged**